ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -459,8 +483,11 @@ public void replicate() {
             // Skip stores that were stopped during call to getReplicaMetadataResponse
             if (!remoteReplicaInfo.getLocalStore().isStarted()) {
               exchangeMetadataResponseList.add(new ExchangeMetadataResponse(ServerErrorCode.Temporarily_Disabled));
-              remoteReplicaInfo.setExchangeMetadataResponse(
-                  new ExchangeMetadataResponse(ServerErrorCode.Temporarily_Disabled));
+              if (replicatingFromRemoteColo && replicationConfig.replicationModelType.equals(
+                  ReplicationModelType.LEADER_BASED) && replicationEngine instanceof ReplicationManager) {
```

Comment:
Are these instanceof checks needed? Maybe the ReplicationEngine implementation can check during construction if LEADER_BASED replication is supported and throw an exception if it is misconfigured.